### PR TITLE
Fix broken encoding (fixes #920)

### LINKED ISSUE #920: Readme downloaded with wrong encoding
Downloading this readme results in a string with the wrong encoding (should be `UTF-8`).

```ruby
client.readme('ScriptFUSION/Porter', accept: 'application/vnd.github.raw')
```

>encoding = ASCII-8BIT

This problem manifests itself when trying to parse it with a markdown processor such as Kramdown.

```ruby
Kramdown::Document.new(
  client.readme('ScriptFUSION/Porter', accept: 'application/vnd.github.raw')
)
```

>Error:  "\xE2" from ASCII-8BIT to UTF-8

### PR DESCRIPTION
**Pull request by lparry**
We have run into problems where using `client.content()` returns an 8-bit ascii string, while the source file is UTF-8, as is reaffirmed by the charset within the content type header.

This PR checks for that charset header, and when found calls `#force_encoding` on the ascii string to make the UTF-8 characters be interpreted correctly.

I had to do some VCR cassette doctoring in order to reference the new fixture as though it was already in the canonical repo and already on master. Hopefully I didn't mess it up in some unforeseen way

Should fix #920

### CODE REVIEW

**Comment by lparry** [comment]
Please let me know if there's anything I can do to help get this merged. Thanks!

**Comment by daniel-beard** [comment]
Am also hitting this issue, is there any path to merging this PR?

**Comment by lparry** [comment]
:tumbleweeds:

**Comment by CarlaBehan1212** [comment]
Thanks for your help with this matter is greatly appreciated and I will get it from the working class people who have been trying to get in touch with me and well-being.

**Comment by eneko** [comment]
Please fix. We've been having this issue for a long time, as Danger relies on this library for diffing pull requests.

**Comment by lparry** [comment]
Hey @hmharvey @tarebyte, I see two releases have happens since this was opened, but this fix has not been included.

Can you please let me know what needs to happen to make this PR acceptable to be merged, so that our team can stop using our own fork?

**Comment by tarebyte** [comment]
@lparry apologies for the delay. After conferring with a few colleagues to verify my initial reaction, we've come to the conclusion that this is a breaking change. We're working toward the release of v5 but I think it'll be a while before we get there with the generated client.

I'm still not entirely sure why this is always returning 8-bit ascii string so I will probably need to track that down. Perhaps it's something to do with the Sawyer/Faraday combo.

In the mean time we can still work together to make this change more robust for all of our use cases. Here's probably what we'll need to do:

- [ ] Reach out to GitHub Support and find out all of the various types of formats that are sent back.
- [ ] Add tests covering all of those cases (and solve the Base64 case as you mentioned)
- [ ] Add a warning to let folks know that the response format will be changing in a future version.

If this sounds agreeable please let me know @lparry, and again thanks for being so patient.

**Comment by tarebyte** [comment]
@lparry thinking about this more, this sounds like a good middleware that we could allow users to opt into. I don't think that would be a breaking change.

**Comment by titanium-cranium** [comment]
Hi @tarebyte:  Unfortunately, @lparry has recently moved on to another company and I'm now picking up the ball to determine the status of the proposed fix.  Any progress?

**Comment by tarebyte** [comment]
> I'm now picking up the ball to determine the status of the proposed fix. Any progress?

I haven't at this point. I might try and look at it in the next few weeks. It's GitHub Satellite this week and I'm on vacation the next.

**Comment by titanium-cranium** [comment]
> I haven't at this point. I might try and look at it in the next few weeks. It's GitHub Satellite this week and I'm on vacation the next.

Beauty! Many thanks.  🙏

**Comment by lucascaton** [comment]
Hi @tarebyte! Unfortunately, @titanium-cranium has recently moved on to another company and I'm now picking up the ball to determine the status of the proposed fix. #dejavu 😅

Is there any news on this? Could you please review and merge this change? Thank you!

**Comment by nickfloyd** [comment]
'👋 Hey Friends, this pull request has been marked as stale because it has no recent activity. It will be closed if no further activity occurs. Please add the `Status: pinned` label if this issue needs to remain open/active. Thank you for your contributions and help in keeping things tidy!'

**Comment by lucascaton** [comment]
Hey @nickfloyd! 👋  Why not merge it, instead?

We've been using a fork with this solution and it's been working well since 2019 (when this PR was opened).

**Comment by nickfloyd** [comment]
Hey @lucascaton, thanks for the response; we can do that!  Being over a year old with merge conflicts will make that a bit challenging if we do not have push access to `envato:lparry/fix-broken-encoding`.  Would you be interested in resolving those conflicts so we can give this a proper review and potentially get it merged in?

If not, no worries; we'll prioritize it and see if we can reach out to @lparry to enable edits for maintainers if needed, if not we can hoist these changes into another more recent branch of main and get it in that way.

**Comment by lucascaton** [comment]
Hi @nickfloyd, I've just merged `main` into this branch, it should be all good to review now 🙂 Thanks!

File: lib/octokit/connection.rb [hunk 1]
```diff
@@ -206,5 +206,14 @@ def parse_query_and_convenience_headers(options)
 
       opts
     end
+
+    def response_data_correctly_encoded(response)
+      # TODO: I'm guessing base64 content responses are still broken, but this fixes the bug we're running into
```
Comment (kfcampbell) [inline lib/octokit/connection.rb:211]: Would you mind removing this TODO comment and filing an issue instead?

Comment (lucascaton) [reply lib/octokit/connection.rb]: Done 🙂 c9aa273

**Comment by lucascaton** [comment]
Hi @nickfloyd, is there anything else you'd like me to change? Thanks!

**Pull request merged**